# Analysis of International Debt statistics

<p>In this little reserch, I am going to analyze The World Bank's international debt data. To provide this research I will use SQL and Jypiter Notebook. This dataset contains information about the amount of debt (in USD) owed by developing countries across several categories (<a href=\"https://www.worldbank.org/en/home\">The World Bank site</a>). We are going to see: </p>
<ul>
<li>What is the total amount of debt that is owed by the countries</li>
<li>Which different debt indicators world bank uses</li>
<li>Which country owns the maximum amount of debt and what does that amount look like?</li>
<li>What is the average amount of debt owed by countries across different debt indicators?</li>
</ul>
<p><img src="img/worlddebt.jpeg" alt></p>


# I. Structure of dataset

In [6]:
%%sql
postgresql:///international_debt
    
    SELECT *
    FROM international_debt
    LIMIT 30;

30 rows affected.


[('Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('72894453.700000003')),
 ('Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('53239440.100000001')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('61739336.899999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('49114729.399999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('39903620.100000001')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (AMT, current US$)', 'DT.AMT.MLAT.CD', Decimal('39107845')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (DIS, current US$)', 'DT.DIS.MLAT.CD', Decimal('23779724.300000001')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (INT, current US$)', 'DT.INT.MLAT.CD', Decimal('13335820')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (AMT, current US$)', 'DT.AMT.OFFT.CD', Decimal('100847181.900000006')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (DIS, current US$)', 'DT.DIS.OFFT.CD', Decimal('72894453.700000003')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (INT, current US$)', 'DT.INT.OFFT.CD', Decimal('53239440.100000001')),
 ('Afghanistan', 'AFG', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('100847181.900000006')),
 ('Albania', 'ALB', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('317194512.5')),
 ('Albania', 'ALB', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('165602386.900000006')),
 ('Albania', 'ALB', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', 'DT.INT.DPNG.CD', Decimal('87884000')),
 ('Albania', 'ALB', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('54250280.600000001')),
 ('Albania', 'ALB', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('76050616.099999994')),
 ('Albania', 'ALB', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('13847333.6')),
 ('Albania', 'ALB', 'PPG, bonds (AMT, current US$)', 'DT.AMT.PBND.CD', Decimal('0')),
 ('Albania', 'ALB', 'PPG, bonds (INT, current US$)', 'DT.INT.PBND.CD', Decimal('31030688.199999999')),
 ('Albania', 'ALB', 'PPG, commercial banks (AMT, current US$)', 'DT.AMT.PCBK.CD', Decimal('39445139.5')),
 ('Albania', 'ALB', 'PPG, commercial banks (DIS, current US$)', 'DT.DIS.PCBK.CD', Decimal('4542664.9')),
 ('Albania', 'ALB', 'PPG, commercial banks (INT, current US$)', 'DT.INT.PCBK.CD', Decimal('4618504.3')),
 ('Albania', 'ALB', 'PPG, multilateral (AMT, current US$)', 'DT.AMT.MLAT.CD', Decimal('182197616.699999988')),
 ('Albania', 'ALB', 'PPG, multilateral (DIS, current US$)', 'DT.DIS.MLAT.CD', Decimal('234321242.300000012')),
 ('Albania', 'ALB', 'PPG, multilateral (INT, current US$)', 'DT.INT.MLAT.CD', Decimal('28101536.100000001')),
 ('Albania', 'ALB', 'PPG, official creditors (AMT, current US$)', 'DT.AMT.OFFT.CD', Decimal('236447897.300000012')),
 ('Albania', 'ALB', 'PPG, official creditors (DIS, current US$)', 'DT.DIS.OFFT.CD', Decimal('310371858.399999976')),
 ('Albania', 'ALB', 'PPG, official creditors (INT, current US$)', 'DT.INT.OFFT.CD', Decimal('41948869.700000003')),
 ('Albania', 'ALB', 'PPG, other private creditors (AMT, current US$)', 'DT.AMT.PROP.CD', Decimal('170018.4'))]

# II. Data of how many contries perfermed in the research.

In [10]:
%%sql
SELECT 
    COUNT(DISTINCT country_name) AS total_amount_of_countries
FROM international_debt;

 * postgresql:///international_debt
1 rows affected.


[(124,)]

## III. Which debt indecators World Bank uses

<p> We can see in the table that there are only 124 while the number of rowes in the dataset is larger. It happens because Worl Bank uses for statistics different debt indecators. </p>
<p> For example, DT.AMT.BLAT.CD indecator represents bilateral debt includes loans from governments and their agencies (including central banks), loans from autonomous bodies, and direct loans from official export credit agencies. (<a href=\"https://databank.worldbank.org/metadataglossary/international-debt-statistics/series/DT.AMT.BLAT.CD\">DT.AMT.BLAT.CD</a>) </p>
<p> DT.AMT.DLXF.CD represents principal repayments on long-term debt are actual amounts of principal (amortization) paid by the borrower in currency, goods, or services in the year specified. (<a href=\"https://databank.worldbank.org/metadataglossary/world-development-indicators/series/DT.AMT.DLXF.CD\">DT.AMT.DLXF.CD</a>) </p>
<p> Detection of all debt indicators:</p>

In [11]:
%%sql

SELECT DISTINCT indicator_code AS distinct_debt_indicator
FROM international_debt
ORDER BY distinct_debt_indicator



 * postgresql:///international_debt
25 rows affected.


[('DT.AMT.BLAT.CD',),
 ('DT.AMT.DLXF.CD',),
 ('DT.AMT.DPNG.CD',),
 ('DT.AMT.MLAT.CD',),
 ('DT.AMT.OFFT.CD',),
 ('DT.AMT.PBND.CD',),
 ('DT.AMT.PCBK.CD',),
 ('DT.AMT.PROP.CD',),
 ('DT.AMT.PRVT.CD',),
 ('DT.DIS.BLAT.CD',),
 ('DT.DIS.DLXF.CD',),
 ('DT.DIS.MLAT.CD',),
 ('DT.DIS.OFFT.CD',),
 ('DT.DIS.PCBK.CD',),
 ('DT.DIS.PROP.CD',),
 ('DT.DIS.PRVT.CD',),
 ('DT.INT.BLAT.CD',),
 ('DT.INT.DLXF.CD',),
 ('DT.INT.DPNG.CD',),
 ('DT.INT.MLAT.CD',),
 ('DT.INT.OFFT.CD',),
 ('DT.INT.PBND.CD',),
 ('DT.INT.PCBK.CD',),
 ('DT.INT.PROP.CD',),
 ('DT.INT.PRVT.CD',)]

## IV. Total amount of debt for all countries in the dataset

<p> To calculate total amount of debt owened by the countries we need to add up debt results for all raws. We will convert the amount to get result in billions of dollars.</p> 
<p><img src="img/totaldebt.png" alt></p>

In [16]:
%%sql
SELECT 
    ROUND(SUM(debt)/1000000000, 2) AS total_debt_bln
FROM international_debt; 

 * postgresql:///international_debt
1 rows affected.


[(Decimal('3079.73'),)]

## V. Country with the highest debt

<p>Now we can find out which country owns the highs amount of debt. This amount will be the sum of different debt indicators from the dataset.</p>
<p><img src="img/chinadebt.png" alt></p>

In [18]:
%%sql
SELECT 
    country_name, ROUND(SUM(debt)/1000000000, 2) AS total_debt_bln
FROM international_debt
GROUP BY country_name
ORDER BY total_debt_bln DESC
LIMIT 1;

 * postgresql:///international_debt
1 rows affected.


[('China', Decimal('285.79'))]

## VI. Calculation of total debt for each category of debt

<p> Dept indicators have a strong socio-economic meaning. For the better performing of the resaerch lets calculate total amount for each debt indicator(millions in USD).</p> 

In [33]:
%%sql
SELECT 
    indicator_code AS debt_indicator,
    indicator_name,
    ROUND(SUM(debt)/1000000, 2) AS total_debt_per_category
FROM international_debt
GROUP BY debt_indicator, indicator_name
ORDER BY total_debt_per_category DESC
LIMIT 20;

 * postgresql:///international_debt
20 rows affected.


[('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('732203.68')),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', Decimal('407734.35')),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)', Decimal('264701.07')),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('238995.98')),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)', Decimal('203858.98')),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)', Decimal('176762.02')),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', Decimal('147707.31')),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)', Decimal('138214.74')),
 ('DT.DIS.MLAT.CD', 'PPG, multilateral (DIS, current US$)', Decimal('100781.24')),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', Decimal('96412.46')),
 ('DT.AMT.BLAT.CD', 'PPG, bilateral (AMT, current US$)', Decimal('86939.60')),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)', Decimal('74701.05')),
 ('DT.INT.PRVT.CD', 'PPG, private creditors (INT, current US$)', Decimal('70534.54')),
 ('DT.AMT.PCBK.CD', 'PPG, commercial banks (AMT, current US$)', Decimal('61728.97')),
 ('DT.AMT.MLAT.CD', 'PPG, multilateral (AMT, current US$)', Decimal('60767.71')),
 ('DT.INT.PBND.CD', 'PPG, bonds (INT, current US$)', Decimal('55526.60')),
 ('DT.AMT.PROP.CD', 'PPG, other private creditors (AMT, current US$)', Decimal('40332.00')),
 ('DT.INT.OFFT.CD', 'PPG, official creditors (INT, current US$)', Decimal('36911.99')),
 ('DT.INT.BLAT.CD', 'PPG, bilateral (INT, current US$)', Decimal('20019.38')),
 ('DT.INT.MLAT.CD', 'PPG, multilateral (INT, current US$)', Decimal('16892.61'))]

## VII. Calculating of average amout of debt for each category

In [1]:
%%sql
SELECT 
    indicator_code AS debt_indicator,
    indicator_name,
    ROUND(AVG(debt)/1000000, 2) AS total_debt_per_category
FROM international_debt
GROUP BY debt_indicator, indicator_name
ORDER BY total_debt_per_category DESC
LIMIT 30;

UsageError: Cell magic `%%sql` not found.


## VIII. The highest amount of principal repayments
<p>We can see that the indicator <code>DT.AMT.DLXF.CD</code> tops the chart of average debt. This category includes repayment of long term debts. Countries take on long-term debt to acquire immediate capital.
<p>An interesting observation in the above finding is that there is a huge difference in the amounts of the indicators after the second one. This indicates that the first two indicators might be the most severe categories in which the countries owe their debts.</p>
<p>We can investigate this a bit more so as to find out which country owes the highest amount of debt in the category of long term debts (<code>DT.AMT.DLXF.CD</code>). Since not all the countries suffer from the same kind of economic disturbances, this finding will allow us to understand that particular country's economic condition a bit more specifically. </p>

In [29]:
%%sql
SELECT 
    country_name AS country, indicator_name, indicator_code
FROM international_debt
WHERE debt = (SELECT 
                 MAX(debt)
             FROM international_debt
             WHERE indicator_code IN('DT.AMT.DLXF.CD'));

 * postgresql:///international_debt
1 rows affected.


[('China', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD')]

## IX. The most common debt indicator

In [16]:
%%sql
SELECT 
    indicator_code, 
    COUNT(indicator_code) AS indicator_count
FROM international_debt
GROUP BY indicator_code
ORDER BY indicator_count DESC, indicator_code DESC
LIMIT 20;


 * postgresql:///international_debt
20 rows affected.


[('DT.INT.OFFT.CD', 124),
 ('DT.INT.MLAT.CD', 124),
 ('DT.INT.DLXF.CD', 124),
 ('DT.AMT.OFFT.CD', 124),
 ('DT.AMT.MLAT.CD', 124),
 ('DT.AMT.DLXF.CD', 124),
 ('DT.DIS.DLXF.CD', 123),
 ('DT.INT.BLAT.CD', 122),
 ('DT.DIS.OFFT.CD', 122),
 ('DT.AMT.BLAT.CD', 122),
 ('DT.DIS.MLAT.CD', 120),
 ('DT.DIS.BLAT.CD', 113),
 ('DT.INT.PRVT.CD', 98),
 ('DT.AMT.PRVT.CD', 98),
 ('DT.INT.PCBK.CD', 84),
 ('DT.AMT.PCBK.CD', 84),
 ('DT.INT.DPNG.CD', 79),
 ('DT.AMT.DPNG.CD', 79),
 ('DT.INT.PBND.CD', 69),
 ('DT.AMT.PBND.CD', 69)]

## X. Total amount of debt for each country
<p><img src="img/Public_debt_percent_gdp_world_map_(2010).svg" alt></p>

In [18]:
%%sql
SELECT country_name, MAX(debt) AS maximum_debt
FROM international_debt
GROUP BY country_name
ORDER BY maximum_debt

 * postgresql:///international_debt
124 rows affected.


[('Tonga', Decimal('10369910.6')),
 ('Sao Tome and Principe', Decimal('10636848.5')),
 ('Comoros', Decimal('13460035.4')),
 ('Dominica', Decimal('21394158.199999999')),
 ('Samoa', Decimal('21715408.399999999')),
 ('Timor-Leste', Decimal('21799986.399999999')),
 ('Turkmenistan', Decimal('29132060.899999999')),
 ('Solomon Islands', Decimal('30749703.899999999')),
 ('Eritrea', Decimal('31110077.800000001')),
 ('Somalia', Decimal('32985034.300000001')),
 ('St. Vincent and the Grenadines', Decimal('33458473.699999999')),
 ('St. Lucia', Decimal('37459842.200000003')),
 ('Vanuatu', Decimal('38626756')),
 ('Grenada', Decimal('41250138.299999997')),
 ('Guinea-Bissau', Decimal('48265990.200000003')),
 ('Central African Republic', Decimal('53717612.899999999')),
 ('Haiti', Decimal('61444338.399999999')),
 ('Burundi', Decimal('64846467.299999997')),
 ('Fiji', Decimal('66510192.600000001')),
 ('Lesotho', Decimal('80548591.599999994')),
 ('Gambia, The', Decimal('89552885.900000006')),
 ('Sierra Leone', Decimal('98339852.799999997')),
 ('Afghanistan', Decimal('100847181.900000006')),
 ('Belize', Decimal('105609924.900000006')),
 ('Iran, Islamic Rep.', Decimal('105885317')),
 ('Eswatini', Decimal('106108147.599999994')),
 ('Guyana', Decimal('108264665.5')),
 ('Cabo Verde', Decimal('109582071.900000006')),
 ('Malawi', Decimal('132165820.5')),
 ('Botswana', Decimal('137995908')),
 ('Liberia', Decimal('150961031.199999988')),
 ('Algeria', Decimal('171185188.099999994')),
 ('Guinea', Decimal('175441526.199999988')),
 ('Djibouti', Decimal('176522478.199999988')),
 ('Chad', Decimal('207090418.300000012')),
 ('Bhutan', Decimal('213258536.900000006')),
 ('Kosovo', Decimal('232317825.800000012')),
 ('Togo', Decimal('236837922.900000006')),
 ('Congo, Rep.', Decimal('247432137.699999988')),
 ('Syrian Arab Republic', Decimal('307261900')),
 ('Maldives', Decimal('324159093.300000012')),
 ('Rwanda', Decimal('327248280.199999988')),
 ('Congo, Dem. Rep.', Decimal('370323229.600000024')),
 ('Yemen, Rep.', Decimal('402113863')),
 ('Honduras', Decimal('412459703.300000012')),
 ('Madagascar', Decimal('446922027.399999976')),
 ('Zimbabwe', Decimal('461632253.699999988')),
 ('Tajikistan', Decimal('467153429.199999988')),
 ('Gabon', Decimal('519921205.699999988')),
 ('Mauritania', Decimal('587173043.899999976')),
 ('Mali', Decimal('602617562.299999952')),
 ('Niger', Decimal('613393210.200000048')),
 ('Burkina Faso', Decimal('617436227.200000048')),
 ('Benin', Decimal('648444772.899999976')),
 ('Montenegro', Decimal('650720957.899999976')),
 ('Jamaica', Decimal('661568432.200000048')),
 ('Sudan', Decimal('704940261.799999952')),
 ('Mozambique', Decimal('749659443.700000048')),
 ('Moldova', Decimal('769232867')),
 ('Cambodia', Decimal('775955981.299999952')),
 ('Albania', Decimal('790248675.200000048')),
 ('Kyrgyz Republic', Decimal('837442293.399999976')),
 ('Armenia', Decimal('845630723.799999952')),
 ('Nepal', Decimal('854955604')),
 ('Macedonia, FYR', Decimal('875846538.5')),
 ("Cote d'Ivoire", Decimal('1081936422.799999952')),
 ('Nicaragua', Decimal('1185575296.799999952')),
 ('Myanmar', Decimal('1246429044.299999952')),
 ('Uganda', Decimal('1359705707.599999905')),
 ('Tanzania', Decimal('1388218481.400000095')),
 ('Bolivia', Decimal('1421491255.200000048')),
 ('Zambia', Decimal('1502399702.799999952')),
 ('Azerbaijan', Decimal('1513813661.400000095')),
 ('Paraguay', Decimal('1528508573.599999905')),
 ('Lao PDR', Decimal('1605959990.099999905')),
 ('Senegal', Decimal('1661790101.599999905')),
 ('Ghana', Decimal('1689328833.5')),
 ('Mauritius', Decimal('1757803962.200000048')),
 ('Costa Rica', Decimal('1776491747.900000095')),
 ('Bulgaria', Decimal('1883205166.200000048')),
 ('Dominican Republic', Decimal('1978111141.200000048')),
 ('Jordan', Decimal('1990343479.200000048')),
 ('Uzbekistan', Decimal('2097517190.900000095')),
 ('Bosnia and Herzegovina', Decimal('2112748020.900000095')),
 ('Tunisia', Decimal('2422062635.800000191')),
 ('El Salvador', Decimal('25745